src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -382,6 +384,17 @@ private static ILambdaExpression CreateBoundLambdaOperation(BoundLambda boundLam
             return new LazyLambdaExpression(signature, body, isInvalid, syntax, type, constantValue);
         }
 
+        private static ILocalFunctionStatement CreateBoundLocalFunctionStatementOperation(BoundLocalFunctionStatement boundLocalFunctionStatement)
+        {
+            IMethodSymbol localFunctionSymbol = boundLocalFunctionStatement.Symbol;
+            Lazy<IBlockStatement> body = new Lazy<IBlockStatement>(() => (IBlockStatement)Create(boundLocalFunctionStatement.Body));
```

Comment:
> Can we assert that the body is in fact a block statement before the lazy, that way if it ever fails it'll fail in a helpful location with a better stack?

It looks like Body is statically typed as BoundBlock.

---
In reply to: [121546721](https://github.com/dotnet/roslyn/pull/20177#discussion_r121546721) [](ancestors = 121546721)